api/models/spell.py

```diff
@@ -0,0 +1,229 @@
+"""
+The model for a spell.
+
+Also includes importing and some presentation logic.
+"""
+
+from django.db import models
+from django.template.defaultfilters import slugify
+from django.core.validators import MinValueValidator
+from django.core.validators import MaxValueValidator
+
+from .models import GameContent
+from ..validators import spell_school_validator
+
+
+class Spell(GameContent):
+    """The model for a spell."""
+
+    # Stored metadata about the spell
+    page = models.TextField(
+        help_text='Page number reference for the document.')
+
+    spell_level = models.IntegerField(
+        validators=[MinValueValidator(0), MaxValueValidator(9)],
+        help_text='Integer representing the level of the spell. Cantrip is 0.')
+
+    dnd_class = models.TextField(
+        help_text='List of classes (comma separated) that can learn this spell.')
+
+    school = models.TextField(
+        validators = [spell_school_validator],
+        help_text='Representation of the school of magic, such as "illusion" or "evocation".')
+
+    # Computed metadata about the spell
+    def v1_level(self):
+        """Presents the spell level in a friendly format."""
+        spell_level_name_lookup = [
+            'Cantrip',
+            '1st-level',
+            '2nd-level',
+            '3rd-level',
+            '4th-level',
+            '5th-level',
+            '6th-level',
+            '7th-level',
+            '8th-level',
+            '9th-level'
+        ]
+        return spell_level_name_lookup[self.spell_level]
+
+    # Stored data about casting the spell
+    casting_time = models.TextField(
+        help_text='Amount of time it takes to cast the spell, such as "1 bonus action" or "4 hours".')
+
+    range = models.TextField(help_text='Text description of the target range.')
+
+    target_range_sort = models.IntegerField(
+        validators=[MinValueValidator(0)],
+        help_text='Sortable distance ranking to the target.')
+
+    requires_verbal_components = models.BooleanField(
+        help_text='Casting this spell requires verbal components.')
+    requires_somatic_components = models.BooleanField(
+        help_text='Casting this spell requires somatic components.')
+    requires_material_components = models.BooleanField(
+        help_text='Casting this spell requires material components.')
+
+    material = models.TextField(
+        help_text='Description of the material required.')
+
+    higher_level = models.TextField(
```

Comment:
I think this is nullable.

Reply:
I don't believe so.
https://docs.djangoproject.com/en/4.2/ref/models/fields/#null
Some of this gets weird because we ignore the CUD of CRUD, so while it may be nullable, it's impossible to update to that value.